src/test/java/com/opipo/codechallenge/service/impl/TransactionServiceImplTest.java

```diff
@@ -0,0 +1,331 @@
+    }
+
+    @Test
+    @DisplayName("Given accountIban and no sort then return list with all elements")
+    void givenAccountIbanAndNoSortThenGetList() {
+        String accountIban = "myIban";
+        SortType sortType = null;
+        Transaction transaction1 = Mockito.mock(Transaction.class);
+        TransactionEntity transactionEntity1 = Mockito.mock(TransactionEntity.class);
+        Transaction transaction2 = Mockito.mock(Transaction.class);
+        TransactionEntity transactionEntity2 = Mockito.mock(TransactionEntity.class);
+
+        List<TransactionEntity> transactionEntities = Arrays.asList(transactionEntity1, transactionEntity2);
+
+        Mockito.when(transactionRepository.findAllByAccountIban(accountIban)).thenReturn(transactionEntities);
+
+        Mockito.when(modelMapper.map(transactionEntity1, Transaction.class)).thenReturn(transaction1);
+        Mockito.when(modelMapper.map(transactionEntity2, Transaction.class)).thenReturn(transaction2);
+
+        List<Transaction> actual = transactionService.list(accountIban, sortType);
+
+        assertNotNull(actual);
+        assertFalse(actual.isEmpty());
+        assertTrue(actual.contains(transaction1));
+        assertTrue(actual.contains(transaction2));
+    }
+
+    @Test
+    @DisplayName("Given no accountIban and no sort then return list with all elements")
+    void givenNoAccountIbanAndNoSortThenGetList() {
+        String accountIban = null;
+        SortType sortType = null;
+        Transaction transaction1 = Mockito.mock(Transaction.class);
+        TransactionEntity transactionEntity1 = Mockito.mock(TransactionEntity.class);
+        Transaction transaction2 = Mockito.mock(Transaction.class);
+        TransactionEntity transactionEntity2 = Mockito.mock(TransactionEntity.class);
+
+        List<TransactionEntity> transactionEntities = Arrays.asList(transactionEntity1, transactionEntity2);
+
+        Mockito.when(transactionRepository.findAll()).thenReturn(transactionEntities);
+
+        Mockito.when(modelMapper.map(transactionEntity1, Transaction.class)).thenReturn(transaction1);
+        Mockito.when(modelMapper.map(transactionEntity2, Transaction.class)).thenReturn(transaction2);
+
+        List<Transaction> actual = transactionService.list(accountIban, sortType);
+
+        assertNotNull(actual);
+        assertFalse(actual.isEmpty());
+        assertTrue(actual.contains(transaction1));
+        assertTrue(actual.contains(transaction2));
+    }
+
+    @Test
+    @DisplayName("Given accountIban and sortAsc then return sorted list")
+    @MockitoSettings(strictness = Strictness.LENIENT)
+    void givenAccountIbanAndAscSortThenGetList() {
+        givenAccountIbanSortedThenGetList(true);
+    }
+
+    @Test
+    @DisplayName("Given accountIban and sortDesc then return sorted list")
+    @MockitoSettings(strictness = Strictness.LENIENT)
+    void givenAccountIbanAndDescSortThenGetList() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/app/albertinizao/codechallenge/pullRequest?prid=4681930)